src/charm.py

```diff
@@ -950,12 +959,17 @@ def _initialise_replica_set(self, event: StartEvent) -> None:
                 self._peers.data[self.app]["replica_set_hosts"] = json.dumps(
                     [self._unit_ip(self.unit)]
                 )
+
                 logger.info("User initialization")
                 self._init_operator_user()
                 self._init_backup_user()
                 self._init_monitor_user()
-                logger.info("Manage relations")
-                self.client_relations.oversee_users(None, None)
+
+                # in sharding, user management is handled by mongos subordinate charm
+                if self.is_role(Config.REPLICATION):
```

Comment:
Sorry, the current code is correct, because it is external user management.
However internal user management is still not protected and can be handled as separate PR.